azure-iot-hub/azure/iot/hub/iothub_registry_manager.py

```diff
@@ -453,3 +479,424 @@ def delete_module(self, device_id, module_id, etag=None):
+
+    def get_device_registry_statistics(self):
+        """Retrieves the IoTHub device registry statistics.
+
+        :raises: `HttpOperationError<msrest.exceptions.HttpOperationError>`
+            if the HTTP response status is not in [200].
+
+        :returns: The RegistryStatistics object.
+        """
+        return self.protocol.service.get_device_registry_statistics()
+
+    def get_configuration(self, configuration_id):
+        """Retrieves the IoTHub configuration for a particular device.
+
+        :param str configuration_id: The id of the configuration.
+
+        :raises: `HttpOperationError<msrest.exceptions.HttpOperationError>`
+            if the HTTP response status is not in [200].
+
+        :returns: The Configuration object.
+        """
+        return self.protocol.service.get_configuration(configuration_id)
+
+    def create_configuration(self, configuration_id, configuration):
+        """Creates a configuration for devices or modules of an IoTHub.
+
+        :param str configuration_id: The id of the configuration.
+        :param Configuration configuration: The configuration to create.
+
+        :raises: `HttpOperationError<msrest.exceptions.HttpOperationError>`
+            if the HTTP response status is not in [200].
+
+        :returns: Configuration object containing the created configuration.
+        """
+        return self.protocol.service.create_or_update_configuration(configuration_id, configuration)
+
+    def update_configuration(self, configuration_id, configuration, etag):
+        """Updates a configuration for devices or modules of an IoTHub.
+
+        :param str configuration_id: The id of the configuration.
+        :param Configuration configuration: The configuration to create.
+        :param str etag: The etag (if_match) value to use for the update operation.
+
+        :raises: `HttpOperationError<msrest.exceptions.HttpOperationError>`
+            if the HTTP response status is not in [200].
+
+        :returns: Configuration object containing the updated configuration.
+        """
+        return self.protocol.service.create_or_update_configuration(
+            configuration_id, configuration, etag
+        )
+
+    def delete_configuration(self, configuration_id, etag=None):
+        """Deletes a configuration from an IoTHub.
+
+        :param str configuration_id: The id of the configuration.
+        :param Configuration configuration: The configuration to create.
+        :param str etag: The etag (if_match) value to use for the delete operation.
+
+        :raises: `HttpOperationError<msrest.exceptions.HttpOperationError>`
+            if the HTTP response status is not in [200].
+
+        :returns: Configuration object containing the updated configuration.
+        """
+        if etag is None:
+            etag = "*"
+
+        return self.protocol.service.delete_configuration(configuration_id, etag)
+
+    def get_configurations(self, max_count=None):
```

Comment:
>get_configurations [](start = 8, length = 18)

How many will it return by default ?
can that be mentioned in docs ?

Reply:
I did not find documentation about that. I will update the doc if we got to know the default.

---
In reply to: [349339528](https://github.com/Azure/azure-iot-sdk-python/pull/397#discussion_r349339528) [](ancestors = 349339528)